### PR TITLE
Modify MainActivity to contain Fragment Instances

### PR DESCRIPTION
**Pull request by srowhani**
#24

Let me know what you think on this.

### CODE REVIEW

File: app/src/main/java/com/dankideacentral/dic/MainActivity.java [hunk 1]
```diff
@@ -169,4 +188,9 @@ public void onProviderDisabled(String provider) {
             Toast.makeText(getApplicationContext(), e.toString(), Toast.LENGTH_LONG).show();
         }
     }
+
+    @Override
+    public void onListFragmentInteraction(DummyContent.DummyItem item) {
+
```
Comment (ermel272) [inline app/src/main/java/com/dankideacentral/dic/MainActivity.java:192]: Can this be removed? Edit: Read the class and got my answer... Im not sure I understand the purpose of this method though

File: app/src/main/res/layout/fragment_item.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (yourFavoriteSoftwareDeveloper) [inline app/src/main/res/layout/fragment_item.xml]: is it a fragment_item or a list_item ? .. if its a list item, maybe change the file name.

File: app/src/main/java/com/dankideacentral/dic/MainActivity.java [hunk 3]
```diff
@@ -72,9 +80,20 @@ public void failure(TwitterException exception) {
...
+        toggleButton = (Button) findViewById(R.id.toggle);
+        toggleButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                currentFragment = currentFragment instanceof ItemFragment ?
+                        mapFragment :
+                        new ItemFragment();
+                FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+                transaction.replace(R.id.frame_container, currentFragment);
+                transaction.addToBackStack(null);
```
Comment (yourFavoriteSoftwareDeveloper) [inline app/src/main/java/com/dankideacentral/dic/MainActivity.java]: when we implement the back button from the itemFragment to the map we should add the map view (fragment) to the back stack. null is good for now.

File: app/src/main/java/com/dankideacentral/dic/MainActivity.java [hunk 4]
```diff
@@ -72,9 +80,20 @@ public void failure(TwitterException exception) {
...
             }
         });
-        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
-                .findFragmentById(R.id.map);
-        mapFragment.getMapAsync(this);
+        toggleButton = (Button) findViewById(R.id.toggle);
+        toggleButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                currentFragment = currentFragment instanceof ItemFragment ?
```
Comment (yourFavoriteSoftwareDeveloper) [inline app/src/main/java/com/dankideacentral/dic/MainActivity.java]: In the past I've done it this way... http://stackoverflow.com/questions/9294603/get-currently-displayed-fragment (top answer -- using tags) That way the FragmentManger manages which fragment is being displayed.

**Review by yourFavoriteSoftwareDeveloper — COMMENTED** [review]
probably want to change the name of the fragment_item.xml as it isn't a fragment, its a list item view.

File: app/src/main/java/com/dankideacentral/dic/MainActivity.java (hunk 4 above)
Comment (srowhani) [reply app/src/main/java/com/dankideacentral/dic/MainActivity.java]: Cool thanks!

**Pull request merged**